suseconnect/migration.go

```diff
@@ -466,11 +468,106 @@ func sortMigrationProducts(m connect.MigrationPath, installedIDs connect.StringS
 	})
 }
 
+// three-way comparison of editions (EDITION=VERSION[-RELEASE])
+// release part is ignored
+func compareEditions(left, right string) int {
+	// cut off (optional) release part
+	leftParts := strings.Split(left, "-")
+	rightParts := strings.Split(right, "-")
+	// split version into parts
+	leftParts = strings.Split(leftParts[0], ".")
+	rightParts = strings.Split(rightParts[0], ".")
+
+	// right-pad parts with zeros to match length
+	for len(leftParts) < len(rightParts) {
+		leftParts = append(leftParts, "0")
+	}
+	for len(rightParts) < len(leftParts) {
+		rightParts = append(rightParts, "0")
+	}
+
+	// lenghts are equal so we can use one index
+	for i := range leftParts {
+		var l, r int
+		var err error
+		// take and convert i-th part of left
+		if l, err = strconv.Atoi(leftParts[i]); err != nil {
+			panic(err)
```

Comment:
not sure about panic() here. Would just printing a useful error message and exiting be better?

Reply:
ruby version doesn't do any checking explicitly so I assume it would just die with default stack trace.

Reply:
maybe print message and return 0 here, so cleaning up this repo is skipped?